saq/job.py

```diff
@@ -140,11 +143,21 @@ def abort_id(self):
         return f"{ABORT_ID_PREFIX}{self.key}"
 
     def to_dict(self):
-        return {
-            k: v.name if k == "queue" else v
-            for k, v in self.__dict__.items()
-            if v != self.__dataclass_fields__[k].default  # pylint: disable=no-member
-        }
+        result = {}
```

Comment:
Take a critical look here.

The `meta` attribute has a default factory so that it's easy for different plugins to tack on various metadata without having to worry about `None`.

But that makes the logic for removing defaults a little bit more complex. With the way this is currently implemented, we'll end up generating a frivolous UUID for the key every time we call `to_dict`.

I think what we actually want is something like pydantic's `exclude_unset` option, but I don't think that's so easy with dataclasses. At least with how we're using them.

Reply:
i wonder if we should just simplify and remove the pruning of default values

Reply:
Yeah, adding defaults in the serialized job feels cleaner.

I think the main downside would be Redis memory overhead and bandwidth, right?